CONTRIBUTING.rst

```diff
@@ -380,6 +380,56 @@ pull requests from other contributors yourself after having reviewed
 them.
 
 
+Merge/squash guidelines
+-----------------------
+
+When a PR is approved and ready to be integrated to the `main` branch, one has the option to *merge* the commits unchanged, or *squash* all the commits into a single commit.
+
+Here is some guidelines on how to proceeed, based on the state of the PR commit history:
+
+1. Miscellaneous commits:
+
+   * `Implement X`
+   * `Fix test_a`
+   * `Add myself to AUTHORS`
+   * `!fixup Fix test_a`
+   * `Update tests/test_integration.py`
+   * `Update tests/test_integration.py`
+
+   In this case, prefer to use the **Squash** merge strategy: the commit history is a bit messy (not in a derrogatory way, often one just commits changes because they know the changes will eventually be squashed together), so squashing everything into a single commit is best. Prefer to also edit the default GitHub message to add more details and improve on it.
```

Comment:
> Prefer to also edit

Can we use stronger wording here? As in, “You **MUST** clean up the commit message, making sure it contains useful details.”

Reply:
Sure good idea.